desc/objectives/utils.py

```diff
@@ -92,6 +92,36 @@ def factorize_linear_constraints(objective, constraint, x_scale="auto"):  # noqa
         A = A[:, cols]
     assert A.shape[1] == xp.size
 
+    # check for degenerate rows and delete if necessary
+    # augment A with b so that it only deletes actual degenerate constraints
+    # which are duplicate rows of A that also have duplicate entries of b,
+    # if the entries of b aren't the same then the constraints are actually
+    # incompatible and so we will leave those to be caught later.
+    A_augmented = np.hstack([A, np.reshape(b, (A.shape[0], 1))])
+    row_idx_to_delete = np.array([], dtype=int)
+    for row_idx in range(A_augmented.shape[0]):
+        # find all rows equal to this row
+        rows_equal_to_this_row = np.where(
+            np.all(A_augmented[row_idx, :] == A_augmented, axis=1)
+        )[0]
+        # find the rows equal to this row that are not this row
+        rows_equal_to_this_row_but_not_this_row = rows_equal_to_this_row[
+            rows_equal_to_this_row != row_idx
+        ]
+        # if there are rows equal to this row that aren't this row, AND this particular
+        # row has not already been detected as a duplicate of an earlier one and slated
+        # for deletion, add the duplicate row indices to the array of
+        # rows to be deleted
+        if (
```

Comment:
With this logic do we prevent deleting both of the constraints? Let's say 2nd row and 6th row of augmented A are the same, then when we run the for loop at second row, we will add 6 and for 6th row, we will add 2? So, both of them are deleted? This would probably cause a failing test but I want to be sure.

Reply:
Oh I see, it takes this into account